sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPermissionProperties.java

```diff
@@ -45,25 +57,102 @@ public CosmosPermissionProperties setId(String id) {
      */
     CosmosPermissionProperties(String jsonString) {
         this.permission = new Permission(jsonString);
+        this.permissionName = permission.getId();
+        this.permissionMode = permission.getPermissionMode();
+        this.resourcePartitionKey = permission.getResourcePartitionKey();
+        this.resourceToken = permission.getToken();
+
+        String[] parts = Utils.trimBeginningAndEndingSlashes(permission.getResourceLink()).split("/");
```

Comment:
please use `StringUtils.split` instead of `String.split`

String.split uses regex and is not performant

Reply:
Fixed